dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -138,10 +149,41 @@ public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilde
         {
             throw new Exception("IRecordBuilder does not have a valid record definition.");
         }
+        if (resumeInfo == null)
+        {
+            this.inputStream = new RowServiceInputStream(this.dataPartition, this.originalRecordDef, projectedRecordDefinition, connectTimeout, limit, createPrefetchThread, readSizeKB);
+            this.binaryRecordReader = new BinaryRecordReader(this.inputStream);
+            this.binaryRecordReader.initialize(this.recordBuilder);
+        }
+        else
+        {
+            RowServiceInputStream.RestartInformation restartInfo = new RowServiceInputStream.RestartInformation();
+            restartInfo.streamPos = resumeInfo.inputStreamPos;
+            restartInfo.tokenBin = resumeInfo.tokenBin;
+
+            this.inputStream = new RowServiceInputStream(this.dataPartition, this.originalRecordDef, projectedRecordDefinition, connectTimeout, limit, createPrefetchThread, readSizeKB, restartInfo);
+            long bytesToSkip = resumeInfo.recordReaderStreamPos - resumeInfo.inputStreamPos;
+            if (bytesToSkip < 0)
+            {
+                throw new Exception("Unable to restart unexpected stream pos in record reader.");
```

Comment:
would it be helpful at all to output the stream pos as part of the exception?

Reply:
Good question, I don't think it would help track down why the resume info wasn't valid. This is essentially a sanity check